prae/src/core.rs

```diff
@@ -101,32 +101,50 @@ where
         self.0
     }
 
+    /// Verifies invariants. This is guaranteed to succeed unless you've used
+    /// one of the `unsafe` methods that require variants to be manually upheld.
+    pub fn verify(&self) -> Result<(), E> {
+        G::validate(&self.0).map_or(Ok(()), Err)
+    }
+}
+
+#[cfg(feature = "unchecked-access")]
+trait UncheckedAccess {
```

Comment:
Now I wonder if there's a need for a trait. Seems like adding another `impl` block guarded by the feature gate is enough. This code works fine:
```rust
struct S;

impl S {
    fn a() {}
}

impl S {
    fn b() {}
}

fn main() {
    S::a();
    S::b();
}
```

Reply:
It will be just much cleaner in my opinion

Reply:
My thinking here was that given the unchecked features are quite specialist, I want the user to explicitly import them into the scope they want to use them in. Maybe it's a little over the top? I was imagining the case where I want to know exactly when in my source code the unchecked functions are used. But this can easily be done with a *_unchecked search across the code base.

Let's go with simplicity here, I'll remove the trait.

Reply:
I've moved the unchecked fns into the impl and deleted the `UncheckedAccess` trait.